helm/prefect-server/values.yaml

```diff
@@ -0,0 +1,273 @@
+# prefect-server configuration
+
+# image provides configuration for all prefect images
+# this notably excludes the hasura image
+image:
+  version: "latest"
+  pullPolicy: IfNotPresent
```

Comment:
Why add shared defaults for all prefect images? IMO this adds extra complications that we may not need. A user is likely to upgrade the helm chart (which will bump the release), or want to explicitly set things for specific deployments (say pin the UI to a version), I don't see the need for a shared setting like this.

That said, if we keep this we should change the `image` field to something more descriptive like `imageDefaults` or something. But I think we should drop this for now - can always add it back later if people ask for it.

Reply:
So, I'm presuming that we will not be using "latest" here but pin to a version on release. If we do want to use "latest", the following argument doesn't really apply and I'm happy to move them all into service specific config.

The shared defaults exist because we generally release all of the images together and I don't think that it makes sense to update ~5 image tags each time we bump the helm version. If the user wants to pin the UI that's still straight-forward and I think it's easier for them to say:
- I want to pin these two versions so I will change them within the per-service configs
- The rest I want to leave so those values will remain empty

Otherwise when they pin a couple versions it'll get confusing to merge with updates from us.

Reply:
> The shared defaults exist because we generally release all of the images together and I don't think that it makes sense to update ~5 image tags each time we bump the helm version.

This could be automated, and is also easier to understand (IMO) than a hierarchy of version settings. I'd rather optimize for user usability than shortness of diffs for us.

> Otherwise when they pin a couple versions it'll get confusing to merge with updates from us.

I don't follow. Without a global setting, they'd pin the UI image tag, and leave the others blank (which would default to the versions shipped with the helm chart). When they want to upgrade, they'd upgrade the helm chart and the unpinned images would upgrade as intended.

I don't think it's a good idea to encourage users to upgrade the prefect server without also bumping the chart - some new server versions may require a change to the chart (new config, new secrets, etc...). Exposing the image name and tag is mostly useful IMO for users who want to build their own images, we should still release a new chart with every prefect-server release IMO.

---

That said, if you want to go with your version I'd suggest:

- Remove the global `pullPolicy`, let users set that per-component only
- Remove the nesting and rename as `prefectVersion`, with a comment that this will be used for the `tag` for all prefect images unless overwritten per component.

I found the `image` grouping confusion, since it wasn't fully specifying an image, only defaults for each sub-components. Simplifying to the above should satisfy your goals but is less confusing to me.

Reply:
Yeah I agree the `image` grouping is poorly named. I am a bit confused by this whole bit and presume it's because I've not done any helm upgrading of charts -- I'll just move to per service image settings and leave them at 'latest'  by default and we can write automation to pin them to a version in the future.